presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -212,6 +213,23 @@ private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> s
                 DoubleStream::max);
     }
 
+    private Estimate calculateDataSize(Map<String, PartitionStatistics> statisticsByPartitionName, String column, Estimate rowCount)
+    {
+        return summarizePartitionStatistics(
+                statisticsByPartitionName.values(),
+                column,
+                hiveColumnStatistics -> {
+                    if (rowCount.isValueUnknown() || !hiveColumnStatistics.getAverageColumnLength().isPresent()) {
+                        return OptionalDouble.empty();
+                    }
+
+                    OptionalLong nullCount = hiveColumnStatistics.getNullsCount();
+                    double nonNulls = nullCount.isPresent() ? rowCount.getValue() - nullCount.getAsLong() : rowCount.getValue();
```

Comment:
`double nonNulls = rowCount.getValue() - nullCount.orElse(0);` ?

more importantly, you seem to subtract per-partition value (`hiveColumnStatistics.getNullsCount()`) from per-table value (`rowCount`). Please explain.

Reply:
did not realize that.  Thanks, I'll fix it.